### ISSUE TITLE
Adding SCIELAB/CIELAB to the ISETBio repo?

### ISSUE THREAD

**Issue by tlian7** [isetbio/isetbio#390]
Hi everyone,

I've been doing a lot of analyses with S-CIELAB/CIELAB in combination with ISETBio. Currently these two metrics are located in the ISETCam repository. Unfortunately ISETCam and ISETBio have some conflicts with each other and can't be on the same path. To get around that, I've been extracting the metrics from ISETCam and keeping them in a separate folder. 

This occasionally makes it difficult to share my code. I've been considering moving the metrics into ISETBio to make things easier and more consistent. 

However, I see two minor downsides of doing this; (1) we would have yet another duplicate of a function in ISETBio and ISETCam and (2) these metrics are currently meant to be run on display RGB images, so one could argue whether they truly belong in ISETBio. 

Do you guys have any thoughts on this? It's not a huge deal to have a separate "metrics" folder, but I'm wondering if there's a better way.

**Comment by javiribera** [comment]
I am in the same situation and manually adding separate folders to the Matlab path is a bit inconvenient.

About downside (2): S-CIELAB does not have to take RGB images necessarily, in fact [the current implementation](https://github.com/ISET/isetcam/blob/master/metrics/scielab/scielab.m) accepts both LMS and XYZ images. If the concern is that the metrics do not really fit a toolset meant to model biological vision, one could make a similar argument for ISETCam. In fact, since SCIELAB and CIELAB model properties of the HVS, it seems to me they fit into ISETBio as much as into ISETCam.

I agree on downside (1). I cannot see a workaround other than keeping an eye on both and committing the changes to both repositories, but I still favor this proposal.

**Comment by DavidBrainard** [comment]
ISETBio has an "external" folder into which we put code where the master copy is elsewhere.  For example, there are a number of Psychtoolbox functions in there, that are needed by ISETBio.  And there is a script that I run from time to time that updates the ISETBio version from the Psychtoolbox master, and clobbers any changes made to the routines in ISETBio.  

I am not very fond of this solution, as sometimes someone makes changes to the code in ISETBio and this causes all sorts of headaches.

I think there is some way to make git suck in the code from the repository in a more automated fashion, but I have not figured that out.

The other option is for the world to adopt the ToolboxToolbox (github - toolboxhub organization).  Then all the dependencies can be automatically happen.  But this does not seem too likely anytime soon.

What I do feel strongly about is that there should be one master copy of the code, and some mechanism to keep the other copies roughly in sync with the master.

**Comment by javiribera** [comment]
Is [synchronizeISETBIOWithRepository.m](https://github.com/isetbio/isetbio/blob/master/external/synchronizeISETBIOWithRepository.m) the script you are talking about? This seems like the cleanest possible way for now, since ToolboxToolbox is not a short-term solution. We could use it to bring the entire "isetcam/metrics" into "isetbio/external/metrics". 

This may bring some confusion with [isettools/metrics](https://github.com/isetbio/isetbio/tree/master/isettools/metrics). Maybe we should rename that directory?

**Comment by DavidBrainard** [comment]
Yes, that is the function.  What we'd do is add a new external subdir metrics as you describe, and the right example code in the synchronize script to make it easy to update those.

There isn't much in isettools/metrics.  We might just consider moving that code into isetcam/metrics, if it isn't there already, and then it would all be relatively simple.

**Comment by wandell** [comment]
I am eager to hear from @tlian7 about this.  Her current work also includes SSIM and S-CIELAB, and I would like to make sure that whatever we decide doesn't interfere with the simulations she is running.

**Comment by DavidBrainard** [comment]
Makes sense.
